pass-nihms-loader/nihms-data-transform-load/src/test/java/org/eclipse/pass/loader/nihms/entrez/EntrezPmidLookupTest.java

```diff
@@ -17,42 +17,68 @@
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+import org.apache.commons.io.IOUtils;
 import org.json.JSONObject;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.mock.mockito.MockBean;
 
 /**
  * @author Karen Hanson
- * @version $Id$
  */
 public class EntrezPmidLookupTest {
 
-    @Test
-    public void testGetEntrezRecordJson() {
-        PmidLookup apiService = new PmidLookup();
+    @MockBean
+    protected PmidLookup mockPmidLookup;
 
-        String pmid = "29249144";
+    @BeforeEach
+    public void setUp() {
+        mockPmidLookup = mock(PmidLookup.class);
+    }
+
+    @Test
+    public void testGetEntrezRecordJson() throws IOException {
+        JSONObject entrezJson = new JSONObject(IOUtils.toString(getClass().getClassLoader().
+                getResourceAsStream("pmidrecord.json"), StandardCharsets.UTF_8));
+        String pmid = "11111111";
 
-        JSONObject pmr = apiService.retrievePubMedRecordAsJson(pmid);
-        assertTrue(pmr.getString("source").contains("Proteome"));
+        when(mockPmidLookup.retrievePubMedRecordAsJson(pmid)).thenReturn(entrezJson);
```

Comment:
I decided to go with putting the entrez service url in the application.properties and in the IT, it will be the same endpoint except I need to get the wiremock port real time in the IT. I was going make a setter method in `PmidLookup` class, does this sound like a good plan for you? I can also emulate what you did with making the entrez service url also a system property. I wasn't sure which would be the better way to go. On one hand it could be overkill having multiple ways to configure this URL (sys env and application.properties), but also using a setter method just for an IT doesn't feel right either.

Reply:
I would try the following:

- add `httpPort` to `@WireMockTest` to set the port, so something like `@WireMockTest(httpPort = 9911)`
- set the `entrez.url` property to `http://localhost:9911` in the `test-application.properties`
- set the new `PmidLookup.entrezUrl` attribute using a `@Value` annotation
 
Note that I had to go with the Sys prop solution in journal loader because it is not spring boot enabled.  Also note that using system props is part of standard spring boot property loading: https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#features.external-config .  Not applicable for journal loader, but just an FYI.

Reply:
Okay cool! I thought the port was randomly assigned, but if I can manual set it, all the better! Thanks for the info.

Reply:
Done.